crates/resolver/src/name_server_pool.rs

```diff
@@ -52,11 +52,18 @@ use crate::proto::{
 pub struct NameServerPool<P: ConnectionProvider> {
     state: Arc<PoolState<P>>,
     active_requests: Arc<Mutex<HashMap<Arc<Vec<u8>>, SharedLookup>>>,
+    ttl: Option<TtlInstant>,
 }
 
 #[derive(Clone)]
 pub(crate) struct SharedLookup(Shared<BoxFuture<'static, Option<Result<DnsResponse, ProtoError>>>>);
 
+// Type alias for TTL unit tests to use tokio's time pause/advance
+#[cfg(not(feature = "tokio"))]
+type TtlInstant = std::time::Instant;
+#[cfg(feature = "tokio")]
+type TtlInstant = tokio::time::Instant;
+
```

Comment:
nit: This should be moved so it doesn't split the `SharedLookup` definition and impls.

Reply:
Huh, I thought I also added a comment about this.